src/_posts/2016-12-23-sonarqube.md

```diff
@@ -0,0 +1,148 @@
+
+## Run SonarQube Analysis Locally
+
+When working with appveyor, it always makes sense to test on your own computer first. The feedback is immediate and you iterate very quickly. It takes a lot longer to modify the appveyor.yml file, push it and wait for a build to go through. Also, if it works locally but doesn't work on AppVeyor, you know the problem is a configuration difference between your computer and the AppVeyor environment (eg a different version of msbuild).
+
+Instead of committing SonarQube executables to the repo, we will download them during the build using Chocolatey.
+
+### Install chocolatey
+
+* [Install Chocolatey](https://chocolatey.org/install) from an administrator command prompt / powershell.
+* Close the command prompt
+
+### Install SonarQube MSBuild Runner
+
+* Open a new administrator command prompt / powershell.
+* `choco install "msbuild-sonarqube-runner" -y`
+
+### Analyse and upload to SonarQube
+
+* `MSBuild.SonarQube.Runner.exe begin /k:"**YourUniqueProjectName**" /d:"sonar.host.url=https://sonarqube.com" /d:"sonar.login=**YourSonarQubeToken**"`
+* `"**YourPathToMSBuild**\MSBuild.exe" "**YourProjectName**.sln"`
+* `MSBuild.SonarQube.Runner.exe end /d:"sonar.login=**YourSonarQubeToken**"`
+
+When finished, you will be able to see the results at [sonarqube.com/](https://sonarqube.com/). If it isn't working, make sure you are using MSBuild 14 and [Java 1.8 or later](http://stackoverflow.com/questions/40249947/msbuild-sonarqube-runner-exe-cant-access-https-sonarqube-com). The SonarQube [Getting Started](https://about.sonarqube.com/get-started/) page is excellent if these instructions become out of date.
+
+## Run SonarQube Analysis on AppVeyor
+
+Now that this is working locally, we can run it on AppVeyor.
+
+Add and commit an **appveyor.yml** file to the root of the repository as follows
+
+```yaml
+before_build:
+  - nuget restore
+build_script:
+  - choco install "msbuild-sonarqube-runner" -y
+  - MSBuild.SonarQube.Runner.exe begin /k:"YourUniqueProjectName" /d:"sonar.host.url=https://sonarqube.com" /d:"sonar.login=YourSonarQubeToken"
+  - msbuild "YourProjectName.sln"
+  - MSBuild.SonarQube.Runner.exe end /d:"sonar.login=YourSonarQubeToken"
+```
+
+Again, you can check the results at [sonarqube.com](https://sonarqube.com/).
+
+## Add a SonarQube badge to the repo
+
+There are are variety of [Quality Gate](https://github.com/QualInsight/qualinsight-plugins-sonarqube-badges/wiki/Quality-Gate-status-badges) and [Metrics](https://github.com/QualInsight/qualinsight-plugins-sonarqube-badges/wiki/Measure-badges) badges available.
+
+To add a standard [![Quality Gate](https://sonarqube.com/api/badges/gate?key=SonarQubeNemoOnAppveyor)](https://sonarqube.com/dashboard/index/SonarQubeNemoOnAppveyor) badge, add the following to readme.md.
+
+`[![Quality Gate](https://sonarqube.com/api/badges/gate?key=YourUniqueProjectName)](https://sonarqube.com/dashboard/index/YourUniqueProjectName)`
```

Comment:
Again I could do this but I would rather not. the code block doesn't wrap and its annoying not being able to see the whole thing at once.